src/ZkFold/Symbolic/Data/ByteString.hs

```diff
@@ -250,39 +180,71 @@ instance
     ) => ReverseEndianness wordSize (ByteString n (Interpreter (Zp p))) where
     reverseEndianness (ByteString (Interpreter v)) = ByteString . Interpreter $ reverseEndianness' @wordSize v
```

Comment:
Note here that basically `reverseEndianness` reorders the outputs and that's all
This can be done with an `hmap`